execution-plane/arcon/tests/recovery_tests.rs

```diff
@@ -0,0 +1,238 @@
+            Box::new(Map::<NormaliseElements, i64>::new(&map_fn)),
+            rocks_backend(3),
+        )
+    });
+
+    system
+        .start_notify(&map_node)
+        .wait_timeout(timeout)
+        .expect("map node never started!");
+
+    // Define Window
+    fn window_fn(buffer: &[i64]) -> NormaliseElements {
+        let sum: i64 = buffer.iter().sum();
+        let count = buffer.len() as i64;
+        let avg = sum / count;
+        let data: Vec<i64> = buffer.iter().map(|x| x / avg).collect();
+        NormaliseElements { data }
+    }
+
+    let mut window_state_backend = rocks_backend(2);
+
+    let window: Box<dyn Window<i64, NormaliseElements>> =
+        Box::new(AppenderWindow::new(&window_fn, &mut *window_state_backend));
+
+    let map_node_ref = map_node.actor_ref().hold().expect("Failed to fetch ref");
+    let channel_strategy =
+        ChannelStrategy::Forward(Forward::new(Channel::Local(map_node_ref), NodeID::new(2)));
+
+    let window_node = system.create(move || {
+        Node::<i64, NormaliseElements>::new(
+            2.into(),
+            vec![1.into()],
+            channel_strategy,
+            Box::new(EventTimeWindowAssigner::<i64, NormaliseElements>::new(
+                window,
+                2,
+                2,
+                0,
+                false,
+                &mut *window_state_backend,
+            )),
+            window_state_backend,
+        )
+    });
+    system
+        .start_notify(&window_node)
+        .wait_timeout(timeout)
+        .expect("window never started!");
+
+    // manually sending data, epochs and watermarks
+    fn watermark(time: u64, sender: u32) -> ArconMessage<i64> {
+        ArconMessage::watermark(time, sender.into())
+    }
+
+    fn element(data: i64, time: u64, sender: u32) -> ArconMessage<i64> {
+        ArconMessage::element(data, Some(time), sender.into())
+    }
+
+    fn epoch(epoch: u64, sender: u32) -> ArconMessage<i64> {
+        ArconMessage::epoch(epoch, sender.into())
+    }
+
+    let window_node_ref = window_node.actor_ref();
+    window_node_ref.tell(element(1, 10, 1));
+    window_node_ref.tell(element(2, 11, 1));
+    window_node_ref.tell(element(3, 12, 1));
+
+    window_node_ref.tell(watermark(11, 1));
+    window_node_ref.tell(epoch(1, 1));
+
+    window_node_ref.tell(element(4, 13, 1));
+    window_node_ref.tell(element(42, 14, 1));
+
+    window_node_ref.tell(watermark(14, 1));
+    window_node_ref.tell(epoch(2, 1));
+
+    window_node_ref.tell(element(69, 15, 1));
+
+    window_node_ref.tell(watermark(20, 1));
+    window_node_ref.tell(epoch(3, 1));
+
+    std::thread::sleep(std::time::Duration::from_secs(1));
+
+    system.shutdown().expect("Shutdown failed");
+
+    // check if any of the nodes panicked
+    if window_node.is_faulty() || map_node.is_faulty() || file_sink_node.is_faulty() {
+        return Err(());
+    }
+
+    Ok(())
+}
+
+#[test]
+fn run_test() {
+    let test_directory = "./recovery_test";
```

Comment:
If the directory is not needed after, then we should use tempfile::tempdir. Or if there is a reason for it to remain after the tests, then add it to gitignore.

Reply:
I recommend something like [Fixture](https://github.com/kompics/kompact/blob/a9b1a5b9909343e4992ade96961cfc33b2353c4a/core/src/lib.rs#L289-L337) actually...seems to work rather well.

Reply:
Oh yeah, no reason. When I first wrote it I just wanted to inspect the folder by hand for some reason and forgot to switch to `tempfile`.